Lc

ligongus opened the pull request:
Reviewed major version and minor version to check their difference, and added responses for this difference.
Added responses to explain what .gitignore does.
Corrected a typo: "git remove -v" should be "git remote -v".
Added Table 3 and Figure 3 to Results.Rmd

ligongus committed:
Reviewed major version and minor version, added responses

ligongus committed:
make a change in RMD file

There is a typo in RMD file: 'git remove -v' should be 'git remote -v'

ligongus committed:
Update Results.Rmd

Added Table 3 and Figure 3

bcjaeger commented:
Thanks, Ligong! I will review this afternoon.

ligongus committed:
update Table 1

changed to mean(sd) for continuous variable

bcjaeger commented on doc/results.Rmd:
````diff
@@ -336,18 +356,32 @@ fig
 ## Figure 2: Penguin dimensions (wide version)
 
-```{r}
+```{r, fig.width=8, fig.height=6}
````
same comment as above

ligongus replied:
removed fig.width=8, fig.height=6 and changed the dimension set_fig_portrait().

bcjaeger commented on doc/results.Rmd:
```diff
@@ -255,11 +255,15 @@ A convention worth following is to make each table a sub section. This makes the
 data %>% 
   infuse_dictionary(meta, units = 'descriptive') %>% 
-  tbl_summary(by = 'island', missing = 'no') %>% 
+  tbl_summary(by = 'island', missing = 'no',
+  statistic = list(
+    all_continuous() ~ "{mean} ({sd})",  # Mean and SD for continuous
+    all_categorical() ~ "{n} ({p}%)"  # Custom format for categorical
+  )) %>% 
```
Nice idea! Great use of the `statistic` input and the code is very readable. This is a fantastic contribution🥇

bcjaeger commented on doc/results.Rmd:
````diff
@@ -324,7 +344,7 @@ The default figure margins are set by the `page_long_above()` and `page_wide_abo
 ## Figure 1: Penguin dimensions (normal version)
 
-```{r}
+```{r, fig.width=8, fig.height=6}
````
This causes the figure image to be wider than the page:

![image](https://github.com/user-attachments/assets/c4c0422b-9f7b-4910-a596-8d62145e4a95)

Can you review and fix? To go above and beyond, see if you can modify the `set_fig_portrait()` function inside of the `setup-figures` code chunk to make the portrait figures have the dimension you prefer.

ligongus replied:
removed fig.width=8, fig.height=6 after "{r," and changed the dimension inside set_fig_portrait().

bcjaeger commented on doc/results.Rmd:
````diff
@@ -336,18 +356,32 @@ fig
+```
+
+
 Hang on, how did the figure size change? That is an intended side effect of `page_long_above()` and `page_wide_above()`. Notably, if you ever want to avoid this convenient side effect, you can override it by setting `fig.width = X` and `fig.height = Y` where X and Y are your preferred figure width and height, in inches.
````
This text no longer makes sense with the addition of Figure 3. Can you review and fix? Reviewing version 1.0 will help clarify what the original purpose of this text was.

ligongus replied:
It's deleted.

bcjaeger commented on doc/results.Rmd:
````diff
@@ -336,18 +356,32 @@ fig
+
+```{r, fig.width=8, fig.height=6}
+
+fig3 <- ggplot(data = data) + 
+  aes(x = flipper_length_mm, y = body_mass_g, color = species) + 
+  geom_point()
````
This looks great- thank you!

bcjaeger commented on doc/results.Rmd:
```diff
@@ -336,18 +356,32 @@ fig
 
+\newpage
+
+## Figure 3: Penguin dimensions (Flipper length vs. Body mass)
```
consistent capitalization in the captions
```suggestion
## Figure 3: Penguin dimensions (flipper length vs. body mass)
```

bcjaeger reviewed (changes requested):
Excellent work, Ligong! I was able to run the pipeline on my computer and reproduce your results without any issues. Given this exercise is meant to give you extra practice updating and incorporating changes into your PR, I have some things I'd like you to investigate.

I noticed that Table 2 looks different in version 1.1 versus 1.0. 

In 1.0, it looks like this:

![image](https://github.com/user-attachments/assets/56bbc8e9-20ff-44b2-9a06-5359ab4411ba)

In 1.1 it looks like this:

![image](https://github.com/user-attachments/assets/82f959d3-0c15-473a-864c-f30d3e57178d)

Can you review this and see if you can get the table back to the format it was in? (i.e., highlight the group label rows in grey)

There is a file called changelog.md in the `doc` directory and it is designed to keep a track record of what is changed from version to version of the results document. Can you review this file and then add your changes for version 1.1? E.g.,

```
## Version 1.1

1. first change you made
2. second change you made
3. etc.
```

Please also delete the initial text in that file (it is just there to demonstrate how to document changes)

There are a few other items to check out. I've left comments in the relevant code lines below.

bcjaeger commented on doc/results.Rmd:
````diff
@@ -302,15 +306,31 @@ data %>%
 ```
+\newpage
+
+## Table 3: Characteristics for Penguin Species subgroups
````
Note: If you accept this suggestion, **make sure you run git pull in your local version of the pull request** before you make additional commits to your local work. If you make a commit online but don't sync that commit with your local work, the remote branch and the local branch will diverge!

ligongus committed:
Update doc/results.Rmd

Co-authored-by: Byron <bcjaeger@perisphere-rwe.com>

ligongus committed:
Update doc/results.Rmd

Co-authored-by: Byron <bcjaeger@perisphere-rwe.com>

ligongus committed:
accept the change

ligongus committed:
accept the changes

ligongus committed:
updated results.rmd

modified figure 2 to the original format, changed the dimension inside set_fig_portrait(), deleted dimensions in other parts, made the titles of tables and figures have the same format, deleted the last paragraph in the report

bcjaeger commented on doc/results.Rmd:
```diff
@@ -302,15 +306,31 @@ data %>%
 To make a landscape section, use the block_section code here. Note that the section will be applied to all content prior to the block reaching up to the most recent block.
 
-`r page_wide_above()`
+
```
Deleting this causes an issue in the document. The issue is that we have text describing how to show results in a landscape layout, but then we don't show any results in a landscape layout. Another issue is that the text about landscape results looks like it's a footnote for Table 3. 

![image](https://github.com/user-attachments/assets/f7e0c12d-6b5d-439b-9e97-0447bdae1c55)

bcjaeger reviewed (changes requested):
This is very close, thank you! I left one comment about landscape results below. Also, I didn't see any changes to the `changelog.md` file for version 1.1. Can you review my request from the initial review and let me know if you have any questions about updating that file?
